### ISSUE TITLE
Preferred email format forec to "Plain Text"

### ISSUE THREAD

**Issue by LapinFou** [jobisoft/TbSync#248]
## Your environment

TbSync version: 1.7
Thunderbird version: 60.6.1 (32 & 64 bits)

- [x]  Yes, I have installed the latest available (beta) version from 
https://github.com/jobisoft/TbSync/releases 
and my issue is not yet fixed, I can still reproduce it.

CardDAV server is running on a Synology NAS.
Client are: TbSync on one side and an Apple iPhone7 (iOS 12.2) on the other side.
By default, the field "Preferred email format" (not sure of the translation, since I'm using the French version)  is always "Plain Text".
I cannot change this parameter with my iPhone.
I can change it to "HTML" with TbSync, but as soon as my iPhone update its contact list, the field reverted back to "plain text".

BR,
Seb

**Comment by jobisoft** [comment]
The plain-text setting of Thunderbird is stored in a property called X-MOZILLA-HTML and I do not think your iPhone should be using that. From what you describe, your iPhone is reverting that setting, which I have no control over.

Can you export one of your contacts which shows this reverting and post its raw vCard Data here?

**Comment by LapinFou** [comment]
Hi John,
I'm very sorry, but I switched to CardBooks. Now, everything seems to be OK.
Thanks for your time.

**Comment by GitHub-Lover** [comment]
@jobisoft 

See @LapinFou did switch to CardBooks. For this user, this issue described on the beginning is not important anymore. I have this issue also. The point is this:

- TbSync 2.9 (carddav)
- Sync with own Baikal Server
- Then Thunderbird contact field 'Prefers to receive messages formatted as' is 'Plain Text'
- **This field force (so in my case) that mails will be sent as plain text mail, even when they are written in HTML-Format**

This would be nice:

- In TbSync preference (or on dav-account setting): A field where can be choosen, how the Thunderbird contact field 'Prefers to receive messages formatted as' will be set.

Thank you for your effort and invested time in this project.

**Comment by jobisoft** [comment]
Have you used the SOGo Connector before with that server?

**Comment by GitHub-Lover** [comment]
@jobisoft 

Thank you for your reply. I guess, i didnt SOGo Connector.

To have more clearance on this case I did the following:
- open my carddav-server with browser
- navigate to a vcf file and download it
- open that vcf file with Atom
- there is the following text in the vcf file: ```X-MOZILLA-HTML:FALSE```

This is the point, why Thunderbird overwrite the general preferences. If I use the CardBook addon then this ```X-MOZILLA-HTML:FALSE``` will not over write the general preferences of Thunderbird. This is why @LapinFou have success with CardBook.

It would be great, if there would be a preference in TbSync, where those ```X-..``` 1) information will not have the affect that Thunderbird overwrite the preference if a mail will be sent in Pain/HTML-Format.

1) I have plenty of those ```X-..``` information:
- ```X-EVOLUTION-..```
- ```X-THUNDERBIRD-```
- ```X-MOZILLA-..```

Thank you.

**Comment by jobisoft** [comment]
The issue here is, that your vcard contains the data X-MOZILLA-HTML:FALSE which is a clear indicator, that this contact should be contacted via plain text emails.

I can of course add a preference which ignores this field during sync. However, I would like to understand, where this data is coming from.

What is the history of that addressbook/carddav server? How long have you been using TbSync? What did you use before TbSync to access your carddav addressbook in Thunderbird?

**Comment by GitHub-Lover** [comment]
Thank you for reopen the issue.

Yes, I guess this is the point ```X-MOZILLA-HTML:FALSE``` makes Thunderbird convert sending mail to pain text only.

I guess every user of carddav protocol use a lot of endpoints (like mobile phones, computers with different operating system and so on). Because some of those endpoints / client(-programs) wrote those additional information (```X-..```) to carddav server the data gets unclean there. (in my case: i dont know which client/program/addon[linux:Evolution, iPad, iPhones, Android with DavX and other clients] did write those ```X-MOZILLA-HTML:FALSE``` to my carddav server).

If we have a such preference which ignore all ```X-..``` then TbSync is better for some of us. If others want to use  this e.g. ```X-MOZILLA-HTML:FALSE``` then they just don't choose this preference in TbSync. A pragmatic way to handle this issue. And you know: people love pragmatic solutions :). I'm not a developer, but I guess is not much work for a such feature in TbSync. Thank you very much.

**Comment by jobisoft** [comment]
To add one more thing to this: I would like to understand the cause of this wrong data. Can you answer the following questions?

- How long have you been using TbSync?
- What did you use before TbSync to access your carddav addressbook in Thunderbird?
- Did/do you use Evolution?

**Comment by jobisoft** [comment]
FYI: The value can be changed of course, it is a simple property of your contact:

![grafik](https://user-images.githubusercontent.com/5830621/70932514-35248800-203a-11ea-9214-63408a4d459d.png)

This is not a bug in TbSync. I just OBEY that value which is stored in your contacts. 

I want to find out, why is it set to the wrong value.

**Comment by GitHub-Lover** [comment]
> To add one more thing to this: I would like to understand the cause of this wrong data. Can you answer the following questions?
> 
>     * How long have you been using TbSync?
> 
>     * What did you use before TbSync to access your carddav addressbook in Thunderbird?
> 
>     * Did/do you use Evolution?

- I use TbSync 3 - 4 months
- Before i did use CardBook for sure. Other addons I am unsure. I run my own carddav-server for about 8 years and use Thunderbird since 2005. Beside Thunderbird addons my carddav-server is used by mobile phones (iOS / Android), Evolution for Linux and Contacts in MacOS.
- Yes, i did use Evolution and I will do it in future also.

PS: I was surprised by the fact, that other software add so many information with ```X-..``` tag to my carddav server.

**Comment by GitHub-Lover** [comment]
> FYI: The value can be changed of course, it is a simple property of your contact:
> 
> ![grafik](https://user-images.githubusercontent.com/5830621/70932514-35248800-203a-11ea-9214-63408a4d459d.png)
> 
> This is not a bug in TbSync. I just OBEY that value which is stored in your contacts.
> 
> I want to find out, why it is set to the wrong value.

Yes, this is known by me (also the possibility to put this on contact level). What we discuss here is not a bug (TbSync developer can say: is not my problem!), it is an important enhancement for TbSync users. As I can see in the forum here, I am not the only one, who have this problem. So it is an pragmatic solution for many users. An this solution makes TbSync better!

Thank you!

**Comment by jobisoft** [comment]
I will add the overide option, promised.

Could you do a test for me? 

- Could you create a contact with evolution and see how that contact shows up in TbSync? 
- What is his message format setting in TbSync/Thunderbird?
- Can you alter the message format setting of that contact in evolution? Is that change reflected in TbSync/Thunderbird?
- Can you alter the message format setting of that contact in TbSync/Thunderbird? Is that change reflected in Evolution?

**Comment by jobisoft** [comment]
@GitHub-Lover : Any chance you could do the test with evolution, to see how that impacts the value of `X-MOZILLA-HTML` ?

**Comment by rjanos007** [comment]
I used before SOGo connector. Now I use SOGo server, with TBSync. 
I still have this problem. All my contacts are in Plain. Is there any solution to modify a few thousand addresses, to HTML?
